### PR TITLE
Dockerize dennis

### PR DESCRIPTION
**Pull request by yannispanousis**
Finally.

TODO:

- [x] More testing to make sure the process has no gaps
- [ ] ~~Create cache directory as part of onboarding process in `set-up-my-mac` and `set-up-my-ubuntu`~~ added as part of please scripts
- [x] Add script in lws and in FE repo to call dennis with params: [LWS PR](https://github.com/lystable/lws/pull/1562) and [FE PR](https://github.com/lystable/lystable-frontend/pull/1603)
- [x] Update dennis code to bootstrap git CLI auth with the token
- [x] Update dennis code to set a name and an email... do they need to be correct if the username/passw is?
- [x] Update dennis docs with required params

`docker run --rm -v ~/git/dennis-cache:/git -e REPO=dennis -t dennis release --type minor --user yannispanousis --token $(cat ../go2)`

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+COPY . /dennis
+RUN python setup.py develop
+
+WORKDIR /git
+
+RUN git config --global credential.helper cache && \
+    git config --global credential.helper 'cache --timeout=3600' && \
+    git config --global user.email example@test.com && \
+    git config --global user.name Me
```
Comment (tomek-he-him) [inline Dockerfile]: Any commits pushed by Dennis will have this data attached? If so, how about something like
```sh
    git config --global user.email 'dennis@lystable.com' && \
    git config --global user.name 'Dennis (Bot)'
```

File: README.md [hunk 2]
```diff
@@ -1,32 +1,36 @@
...
-## Install
+## Setup
+
+- It's advised you use a cache directory so Dennis doesn't need to clone the repositories for every release.
+
+```
+mkdir -p ~/.dennis
+```
+
+- You may benefit from either adding a Bash alias, or from creating a one-liner script within your project, for running the releases
```
Comment (tomek-he-him) [inline README.md:11]: How about adding `scripts/dennis` to the LWS and frontend repos? Less steps to explain. “Make work beautiful”.

This way you can `please dennis`, or you can add `scripts/` to your PATH.

File: entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+if [[ -z $REPO ]]; then
+  echo 'You must set the REPO environment variable'
+  exit 1
+fi
+
+if [[ ! -d /git/$REPO ]]; then
+  # Clone the repo if it's not cloned
+  echo "Repo not found, cloning now..."
+  git clone -b develop https://github.com/$OWNER/$REPO.git /git/$REPO > /dev/null
```
Comment (tomek-he-him) [inline entrypoint.sh]: If we want to make sure names from outside the `lystable` namespace don’t clash, we should record the owner name in the cache path. As in 
<pre>
git clone -b develop https://github.com/$OWNER/$REPO.git /git/<b>$OWNER/</b>$REPO > /dev/null
</pre>

`git clone` creates the target path recursively, like `mkdir -p` 😋

**Review by tomek-he-him — COMMENTED** [review]
Great stuff dude! Very big improvement on the stability front.

I think there are bigger wins on the DX front though. The new setup procedure isn’t much less complex than the previous one. Unless of course we cover that stuff in the changes you’re cooking for the other repos 😀

File: Dockerfile (hunk 1 above)
Comment (yannispanousis) [reply Dockerfile]: Yep sure, it's just that we only authorise as the person who has passed in their username and token (this is done inside the Python code), yet we need to set these to something so Git doesn't complain. But I'll update the above for cleanliness!

File: README.md (hunk 2 above)
Comment (yannispanousis) [reply README.md:11]: Yep, that's done in https://github.com/lystable/lws/pull/1562 and https://github.com/lystable/lystable-frontend/pull/1603

The above docs are more for external users :)

Comment (yannispanousis) [reply README.md:11]: We are in fact no longer going to need any setup at all for Dennis. We simply run the script that will exist in the repo

File: entrypoint.sh (hunk 3 above)
Comment (yannispanousis) [reply entrypoint.sh]: good shout!

**Comment by yannispanousis** [comment]
@tomekwi thanks man, do my answers clarify things? There is no setup required from the user. The `Dockerfile` takes care of all the ugly stuff.

**Comment by tomek-he-him** [comment]
Niice dude, amazing work 👌👌👌

**Review by tomek-he-him — APPROVED** [review]

**Comment by yannispanousis** [comment]
@tomekwi thanks man, requested changes have been made

**Pull request merged**